### PR TITLE
allow to customize styles in our centreon frontend components

### PR DESCRIPTION
**Pull request by Med1233**
## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.04.x
- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x (master)

<h2> How this pull request can be tested ? </h2>

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: packages/centreon-ui/src/ContentWithCircularProgress/index.tsx [hunk 1]
```diff
@@ -26,13 +27,14 @@ const ContentWithCircularLoading = ({
   children,
   loadingIndicatorSize = undefined,
   alignCenter = true,
+  className,
 }: Props): JSX.Element => {
   const { classes, cx } = useStyles();
 
   if (loading) {
     return (
       <div className={cx({ [classes.alignCenter]: alignCenter })}>
```
Comment (Thebarda) [inline packages/centreon-ui/src/ContentWithCircularProgress/index.tsx:36]: You can add support for the className here by creating a new prop (`loadingContainerClassname` for instance)

File: packages/centreon-ui/src/Dialog/index.tsx [hunk 2]
```diff
@@ -38,9 +39,16 @@ const Dialog = ({
   confirmDisabled = false,
   cancelDisabled = false,
   submitting = false,
+  className,
   ...rest
 }: Props): JSX.Element => (
-  <MuiDialog open={open} scroll="paper" onClose={onClose} {...rest}>
+  <MuiDialog
```
Comment (Thebarda) [inline packages/centreon-ui/src/Dialog/index.tsx:45]: Add overriding support for className for DialogTitle, DialogContent and DialogActions

**Review by Thebarda — COMMENTED** [review]
Great job 👍

File: packages/centreon-ui/src/Dialog/index.tsx [hunk 3]
```diff
@@ -38,30 +55,50 @@ const Dialog = ({
...
-      {onCancel && (
-        <Button color="primary" disabled={cancelDisabled} onClick={onCancel}>
-          {labelCancel}
-        </Button>
+}: Props): JSX.Element => {
+  const { classes, cx } = useStyles({ contentWidth });
+
+  return (
+    <MuiDialog
+      classes={{ paper: muiDialogclassName }}
```
Comment (Thebarda) [inline packages/centreon-ui/src/Dialog/index.tsx:68]:
```suggestion
```
The classes prop is passed with the spread operator

File: packages/centreon-ui/src/Dialog/index.tsx [hunk 4]
```diff
@@ -10,14 +12,29 @@ import {
...
+  className?: string;
   confirmDisabled?: boolean;
   contentWidth?: number;
+  dialogActionsClassName?: string;
+  dialogContentClassName?: string;
+  dialogTitleClassName?: string;
   labelCancel?: string;
   labelConfirm?: string;
   labelTitle?: string;
+  muiDialogclassName?: string;
```
Comment (Thebarda) [inline packages/centreon-ui/src/Dialog/index.tsx:37]:
```suggestion
  dialogClassName?: string;
```

File: packages/centreon-ui/src/Dialog/index.tsx [hunk 5]
```diff
@@ -10,11 +12,26 @@ import {
...
+
 export type Props = {
   cancelDisabled?: boolean;
   children?: ReactNode;
+  className?: string;
   confirmDisabled?: boolean;
   contentWidth?: number;
+  dialogActionsClassName?: string;
+  dialogContentClassName?: string;
+  dialogPaperclassName?: string;
```
Comment (Thebarda) [inline packages/centreon-ui/src/Dialog/index.tsx:33]:
```suggestion
  dialogPaperClassName?: string;
```

File: packages/centreon-ui/src/Dialog/index.stories.tsx [hunk 6]
```diff
@@ -39,3 +59,24 @@ export const confirmDisabledSubmitting = (): JSX.Element => (
...
+const CustomDialog = (): JSX.Element => {
+  const { classes } = useStyles();
+
+  return (
+    <Dialog
+      open
+      className={classes.root}
+      dialogActionsClassName={classes.actions}
+      dialogContentClassName={classes.content}
+      dialogPaperclassName={classes.paper}
```
Comment (Thebarda) [inline packages/centreon-ui/src/Dialog/index.stories.tsx:72]:
```suggestion
      dialogPaperClassName={classes.paper}
```

File: packages/centreon-ui/src/Dialog/index.tsx [hunk 7]
```diff
@@ -38,30 +55,52 @@ const Dialog = ({
...
-        <Button color="primary" disabled={cancelDisabled} onClick={onCancel}>
-          {labelCancel}
-        </Button>
+}: Props): JSX.Element => {
+  const { classes, cx } = useStyles({ contentWidth });
+
+  return (
+    <MuiDialog
+      PaperProps={{
+        className: dialogPaperclassName,
```
Comment (Thebarda) [inline packages/centreon-ui/src/Dialog/index.tsx:69]:
```suggestion
        className: dialogPaperClassName,
```

**Review by Thebarda — APPROVED** [review]

**Review by JKancel — APPROVED** [review]

**Pull request merged**